clang/test/Driver/sycl-native-cpu.cpp

```diff
@@ -19,7 +19,14 @@
 // CHECK-WIN-NOT: dwarf
 
 // checks that -sycl-opt is not enabled by default on NativeCPU so that the full llvm optimization is enabled
-// RUN:   %clang -fsycl -fsycl-targets=native_cpu -### %s 2>&1 | FileCheck -check-prefix=CHECK-OPTS %s
+// Also check that we pass the expected backend options.
+// RUN:   %clang -fsycl -fsycl-targets=native_cpu -target aarch64-unknown-linux-gnu -march=armv9.4-a -### %s 2>&1 | FileCheck -check-prefix=CHECK-OPTS %s
+// CHECK-OPTS: clang{{.*}}"-triple" "native_cpu"{{.*}}"-aux-triple" "[[TRIPLE:[^"]*]]"
+// CHECK-OPTS: clang{{.*}}"-triple" "[[TRIPLE]]"{{.*}}"-fsycl-is-device"
+// CHECK-OPTS-NOT: -sycl-opt
+// CHECK-OPTS-SAME: "-Wno-override-module" "-mllvm" "-sycl-native-cpu-backend"
+// CHECK-OPTS-NOT: -sycl-opt
```

Comment:
The additional `-sycl-opt` `NOT` checks are not necessary, only need the one on line 26.

Reply:
The addition of `-sycl-opt` only can occur once in the `-cc1` command line, so the first check covers that, there is no chance of `-sycl-opt` showing up after the `-Wno-override-module` addition.

Reply:
Ah, I see what you mean, thanks for the explanation. Done!